AppointmentReminders.Web/Controllers/AppointmentsController.cs

```diff
@@ -1,46 +1,92 @@
 ﻿using System;
-using System.Collections.Generic;
 using System.Linq;
 using System.Net;
-using System.Web;
 using System.Web.Mvc;
 using AppointmentReminders.Web.Models;
+using AppointmentReminders.Web.Models.Repository;
 
 namespace AppointmentReminders.Web.Controllers
 {
     public class AppointmentsController : Controller
     {
-        // GET: Appointments
+        private readonly IAppointmentRepository _repository;
 
-        public ActionResult Index()
+        public AppointmentsController() : this(new AppointmentRepository()) { }
+
+        public AppointmentsController(IAppointmentRepository repository)
+        {
+            _repository = repository;
+        }
+
+        public SelectListItem[] Timezones
         {
-            var appointments = new List<Appointment>
+            get
             {
-               new Appointment{Id = 1, Name = "Frank", PhoneNumber = "+593992670240",Time = new DateTime(), CreatedAt = new DateTime() } 
-            };
+                var systemTimeZones = TimeZoneInfo.GetSystemTimeZones();
+                return systemTimeZones.Select(systemTimeZone => new SelectListItem
+                {
+                    Text = systemTimeZone.DisplayName,
+                    Value = systemTimeZone.Id
+                }).ToArray();
+            }
+        }
+
+        // GET: Appointments
+        public ActionResult Index()
+        {
+            var appointments = _repository.FindAll();
             return View(appointments);
+        }
 
+        // GET: Appointments/Details/5
+        public ActionResult Details(int? id)
+        {
+            if (id == null)
```

Comment:
Yes, we can. We can add some constraints when dealing with routes, I think I can have a look on this.